### PR TITLE
Vania abramov

### PR DESCRIPTION
**Pull request by vanekh-abramov**
**Create** device fetching ### check ###

### CODE REVIEW

File: client/src/components/device/Devices.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import axios from "axios";
+import styles from "../../styles/devices.module.scss";
+import DeviceItem from "./DeviceItem";
+
+const Devices = () => {
+  const [post, setPosts] = useState([]);
+
+  useEffect(() => {
+    axios
+      .get("http://localhost:5000/api/device")
```
Comment (eugeneborodko) [inline client/src/components/device/Devices.jsx]: replace hardcoded url with url from constants file

File: client/src/components/device/Devices.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      .then((res) => setPosts(res.data.rows))
+      .catch((error) => {
+        console.error(error);
+      });
+  }, []);
+
+  console.log(post);
```
Comment (eugeneborodko) [inline client/src/components/device/Devices.jsx]: remove console.log

File: client/src/components/device/DeviceItem.jsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
...f8AFcjln1URl4mM3ve/p+RlTVkluuvXyPkomKRK5GsS6lnz8/2I8oWZgvezUO7KXVpfn+ZdFdwKmlSv/ubf5fkWJ2nEXoADWAAAFTxGos/deu3r0LY57H071b7K79X0IzbGVRtvRb6a8iNCslvJX1uba/dd/hZ/ijXOkmkra8/PchTc43jmT3X3EXiFH2kF3nFKSd46Sst8r5Nq69Su4r2jp4ebjVvCMUm5tWi7/ZhzbW/8Eh8RhJUnB5qc7tTTTV8rcV431/xNEl4CCySh3cl8qjs00k1Jc9t9yTw2j/VTb1dvuS28DGos0Wnz0etreq/Iz4Th3Gokm2tXq72VnzerNjHQgA6JAAAAAAqMfGylHrctyDxCjfW3zyJynjYpKUmkk2m+m38BPxMqk9bSXkzXJrbQ5rYVN76aGtx5eO3mZssOD4Nylnfup6X5tGyC6wtLLCMeiSNoB1cwAAAAAKDiqtKydry0/EvznuKQtPvq8Vr53aIz42NVSTus7Sun8TZmTV73NTs/GL7q5213v87CUstmlu9SFI/EJxc7ZFOrCDlBSaSd7qyb2va1/Epuz08NUj7SLUJ1JqpJJ6Z8lpKCfKOZ7LmWHG+z9PEzhKc6kckXG1ObhmUndqTXL5ufeJ8Do1KcI5cvssrpyjpKNrLuvyVjRZ3SXz5FhwuHfb5JW+P8FNSpNOKcnK12n48k/BLQ6Hhke7fq/wACsWVMABaQAAAAAPjVz6AKzG8ObbcbeRDXCqj5R9WX4J/mN2qaHBl9t+i/Us6VNRSSVkjMGyaZsABoAAAAABW8YoprM/IsjXiKSlFxfMyzcI5ijRyvbuvbXWzN0rrVfEwrKUZRjJbOzatfTZCctLeOvqclt1SqnZL5tuRMUpSjJRdm7K/NJvW391r28bG6TulbTfU25bfPTqaIXDcFKl3czlH7Lk25RWiUW/tbbt8zrKNNRSS2RE4fhrLNJavbw/QnHSRNAAawAAAAAAAAAAAAAAAAAAAAAAABD4hgFUS5STun+vgV0MJUTeaN+jTuXoMsbtSLhspcra313/gsaGBjGzerXzsSgJDYADWAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD//Z' alt={id} />
```
Comment (eugeneborodko) [inline client/src/components/device/DeviceItem.jsx]: replace image src with src from api

File: client/src/components/device/DeviceItem.jsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import styles from "../../styles/devices.module.scss";
+
+const DeviceItem = ({createdAt, id, name, price}) => {
+  return (
+      <li key={createdAt} className={styles.device_item}>
```
Comment (eugeneborodko) [inline client/src/components/device/DeviceItem.jsx]: key prop is not necessary here. Please, remove it

File: client/package.json [hunk 5]
```diff
@@ -6,9 +6,12 @@
     "@testing-library/jest-dom": "^5.16.2",
     "@testing-library/react": "^12.1.3",
     "@testing-library/user-event": "^13.5.0",
+    "axios": "^0.26.0",
+    "node": "^17.5.0",
```
Comment (eugeneborodko) [inline client/package.json]: what is the purpose of node package?

**Comment by vanekh-abramov** [comment]
Hi

File: client/src/components/brand/Brands.jsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import React, { useEffect, useState } from "react";
+import { LOCAL_HOST, API_URL, BRAND_URL } from "../../constants/internalLinks";
+import styles from "../../styles/brands.module.scss";
+import axios from "axios";
+
+const Brands = () => {
+  const [brand, setBrand] = useState([]);
+
+  useEffect(() => {
+    axios
```
Comment (eugeneborodko) [inline client/src/components/brand/Brands.jsx]: create your custom useFetch hook to avoid code repeatability

File: client/src/App.js [hunk 7]
```diff
@@ -1,23 +1,24 @@
...
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div className={styles.container}>
+      <Header />
+      <div className={styles.container_inner}>
+        <Routes>
+          <Route path={"/"} exact element={<Home />} />
```
Comment (eugeneborodko) [inline client/src/App.js]: use array.map for routes. Create AppRouter component

File: client/src/components/Header.jsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+const Header = () => {
+  return (
+    <header className={styles.header}>
+      Header
+      <nav className={styles.nav}>
+        <NavLink className={styles.nav_item} extend="true" aria-current="page" to={"/"}>
+          Home
+        </NavLink>
+        <NavLink className={styles.nav_item} to={"/catalog"}>Catalog</NavLink>
```
Comment (eugeneborodko) [inline client/src/components/Header.jsx]: use constants instead of hardcoded urls. Use array.map for nav links

File: client/src/App.js [hunk 9]
```diff
@@ -1,23 +1,24 @@
-import logo from './logo.svg';
-import './App.css';
+import styles from "../src/styles/app.module.scss"
+// import Devices from "../src/components/device/Devices"
```
Comment (eugeneborodko) [inline client/src/App.js]: remove commented code

File: client/src/components/device/Devices.jsx [hunk 10]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import styles from "../../styles/devices.module.scss";
+import DeviceItem from "./DeviceItem";
+import { LOCAL_HOST, API_URL, DEVICE_URL } from "../../constants/internalLinks";
+import useFetch from "../../hooks/useFetch";
+
+const Devices = () => {
+
+  const [devices, loading, error] = useFetch(`${LOCAL_HOST}/${API_URL}/${DEVICE_URL}?limit=20`);
```
Comment (eugeneborodko) [inline client/src/components/device/Devices.jsx]: limit value shouldn't be hardcoded. Please, replace with a variable

File: client/src/styles/app.module.scss [hunk 11]
```diff
@@ -0,0 +1,8 @@
+.container {
+  margin: 15px auto;
+  max-width: 1440px;
+}
+
+// .container_inner {
```
Comment (eugeneborodko) [inline client/src/styles/app.module.scss]: remove commented code

File: client/src/styles/catalog.module.scss [hunk 12]
```diff
@@ -0,0 +1,22 @@
+.container_inner {
+  display: grid;
+  grid-template-columns: 0.4fr 1.6fr;
+  // grid-template-rows: 0.5fr 1.5fr;
```
Comment (eugeneborodko) [inline client/src/styles/catalog.module.scss]: remove commented code

File: client/src/styles/devices.module.scss [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+    cursor: pointer;
+    transition: background-color 0.7s linear;
+    &:hover {
+      color: v.$base;
+      background-color: v.$second;
+    }
+    img {
+      width: 100%;
+      height: 50%;
+      // border-radius: 1rem 1rem 0 0;
```
Comment (eugeneborodko) [inline client/src/styles/devices.module.scss]: remove commented code

File: client/src/styles/types.module.scss [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+  padding: 1rem;
+  border-radius: v.$bordradius;
+  background-color: v.$base;
+  .types_item {
+    color: v.$second;
+    padding: 0.5rem;
+    font-size: 1.5rem;
+    text-decoration: none;
+    list-style: none;
+    // margin: 1rem 0;
```
Comment (eugeneborodko) [inline client/src/styles/types.module.scss]: remove commented code

**Pull request merged**